### ISSUE TITLE
Add optimizers for multiplication of exponentials

### ISSUE THREAD

**Issue by ajax2leet** [Theano/Theano#5263]
Theano currently doesn't seem to optimize expressions like a**(b) * a**(c) to a**(b+c). Since exponentiation is expensivem this would be a useful optimization to have by default.

**Comment by nouiz** [comment]
Yes, it would be great to add this. Do you want to try to do it? If so, there is documentation and exerices in the advanced part of this tutorial:

https://github.com/abergeron/ccw_tutorial_theano

I do not think the core of Theano will work on this shortly.
